confluent/confluent.go

```diff
@@ -57,13 +89,35 @@ func createBasicConfigMap(properties map[string]string) kafka.ConfigMap {
 	return configMap
 }
 
-// CreateTopic creates a topic using the Admin Client API.
-func CreateTopic(producer *kafka.Producer, topic string) {
+// CreateTopic creates a topic using the Admin Client API with default settings.
+// This function will panic if the topic does not exist and cannot be created
+// for any reason.
```

Comment:
So this is a general question I have about using Kafka. Do we need to create a topic on code? Or do we simply need to point the producer/consumer at an already existing event?

Reply:
That is less a Kafka usage question and more of a practice question. In other words, this question is answered by determining the practice within a particular development team for the kinds of use cases they handle. But the need for the ability to create topics programmatically is unquestionable. That capability is core to the Kafka API extended by this library, and it must not be hidden or taken away.

- In cases where the event types are known ahead of time and there is little fluctuation in consumer processes, it is common for topics to be created administratively first.
- In cases where event types are identified on the fly and consumers appear and disappear continually, it is common for producers and consumers to create topics programmatically at runtime.